client/dwh_migration_client/object_name_mapping.py

```diff
@@ -0,0 +1,111 @@
+# limitations under the License.
+"""Parses the object name mapping json file into ObjectNameMappingList."""
+
+import json
+import logging
+from pprint import pformat
+
+from google.cloud.bigquery_migration_v2 import (
+    NameMappingKey,
+    NameMappingValue,
+    ObjectNameMapping,
+    ObjectNameMappingList,
+)
+from marshmallow import Schema, ValidationError, fields, post_load, validate
+
+
+class NameMappingKeySchema(Schema):
+    """Schema and data validator for NameMappingKey."""
+
+    type = fields.String(
+        required=True,
+        validate=validate.OneOf(
+            (
+                "DATABASE",
+                "SCHEMA",
+                "RELATION",
+                "ATTRIBUTE",
+                "RELATION_ALIAS",
+                "ATTRIBUTE_ALIAS",
+                "FUNCTION",
+            )
+        ),
+    )
+    database = fields.String()
+    schema = fields.String()
+    relation = fields.String()
+    attribute = fields.String()
+
+    @post_load
+    def build(self, data, **kwargs):  # type: ignore[no-untyped-def] # pylint: disable=unused-argument
+        return NameMappingKey(**data)
+
+
+class NameMappingValueSchema(Schema):
+    """Schema and data validator for NameMappingValue."""
+
+    database = fields.String()
+    schema = fields.String()
+    relation = fields.String()
+    attribute = fields.String()
+
+    @post_load
+    def build(self, data, **kwargs):  # type: ignore[no-untyped-def] # pylint: disable=unused-argument
+        return NameMappingValue(**data)
+
+
+class ObjectNameMappingSchema(Schema):
+    """Schema and data validator for ObjectNameMapping."""
+
+    source = fields.Nested(NameMappingKeySchema, required=True)
+    target = fields.Nested(NameMappingValueSchema, required=True)
+
+    @post_load
+    def build(self, data, **kwargs):  # type: ignore[no-untyped-def] # pylint: disable=unused-argument
+        return ObjectNameMapping(**data)
+
+
+class ObjectNameMappingListSchema(Schema):
+    """Schema and data validator for ObjectNameMappingList."""
+
+    name_map = fields.List(fields.Nested(ObjectNameMappingSchema), required=True)
+
+    @post_load
+    def build(self, data, **kwargs):  # type: ignore[no-untyped-def] # pylint: disable=unused-argument
+        return ObjectNameMappingList(**data)
+
+
+def parse(object_name_mapping_file_path: str) -> ObjectNameMappingList:
+    """Parses the object name mapping json file into ObjectNameMappingList."""
+    logging.info("Parsing object name mapping file: %s.", object_name_mapping_file_path)
+    with open(object_name_mapping_file_path, encoding="utf-8") as file:
+        data = json.load(file)
+    try:
+        object_name_mapping_list: ObjectNameMappingList = (
+            ObjectNameMappingListSchema().load(data)
+        )
+    except ValidationError as error:
+        logging.error(
+            "Invalid object name mapping file: %s: %s.",
+            object_name_mapping_file_path,
+            error,
```

Comment:
I am curious what kind of error messages we will get from marshmallow.  In the old logic, we basically just prepared the error message indicating how to fix the error in the json file.

Reply:
Something along the lines of:

Required field `location` not specified:
```
2022-07-07 03:25:56,198: ERROR: Invalid config file:/usr/local/google/home/ajwelch/Documents/code/integrations/dwh-migration/dwh-migration-tools/client/config.yaml: {'translation_config': {'location': ['Missing data for required field.']}}.
```

Invalid fields in the config file (I accidentally pasted an old config form before we moved input and output to cli args):
```
2022-07-07 05:08:08,778: ERROR: Invalid config file: /usr/local/google/home/ajwelch/Documents/code/integrations/dwh-migration/dwh-migration-tools/client/config.yaml: {'translation_config': {'input_directory': ['Unknown field.'], 'output_directory': ['Unknown field.']}}.
```

But we can make these whatever we want.